src/liboslexec/oslexec_pvt.h

```diff
@@ -913,6 +926,9 @@ class ShadingSystemImpl
     mutable std::map<ustring,long long> m_group_profile_times;
     // N.B. group_profile_times is protected by m_stat_mutex.
 
+    // A mapping from strings to integers, used for constructing tagged strings
+    std::map<ustring, uint64_t> m_tag_map;
+
```

Comment:
Is it worth specializing so that key comparison uses `ustring::c_str()` avoiding the `ustring::compare` call to `strcmp`, or is lexicographical sorting needed?